intellij-plugin-structure/structure-intellij/src/main/java/com/jetbrains/plugin/structure/intellij/plugin/PluginCreator.kt

```diff
@@ -892,3 +896,10 @@ internal class PluginCreator private constructor(
       is PluginCreationFail -> this.errorsAndWarnings.filter { it.level === ERROR }
     }
 }
+
+private fun PluginCreationResult<IdePlugin>.add(telemetry: PluginTelemetry): PluginCreationResult<IdePlugin> {
```

Comment:
why don't we add telemetry? it could still contain useful (partial) results?

Reply:
Adding a telemetry to the `PluginCreationFail` is quite a large change. Such creation result does not carry plugin metadata (only `File` and the corresponding descriptor errors). Adding this will complicate this PR.

Note that this will carry only the barebones telemetry - plugin archive size, and descriptor parsing duration.

Let's do that in a subsequent PR.